ocaml/parsing/printast.ml

```diff
@@ -135,18 +135,32 @@ let typevars ppf vs =
     (* Don't use Pprintast.tyvar, as that causes a dependency cycle with
        Jane_syntax, which depends on this module for debugging. *)
 
+let modalities i ppf modalities =
+  line i ppf "modalities\n";
+  list i string_loc ppf (
+    List.map (Location.map (fun (Modality x) -> x)) modalities
+  )
+
+let modes i ppf modes =
+  line i ppf "modes\n";
+  list i string_loc ppf (
+    List.map (Location.map (fun (Mode x) -> x)) modes
+  )
+
```

Comment:
It might be marginally better, if we define `modality` and `mode`, and define the plural version separetely.

Reply:
Also, the printintg might be cleaner, if we don't print anything if `modes` is empty. Similar to the the printing of `attributes`.